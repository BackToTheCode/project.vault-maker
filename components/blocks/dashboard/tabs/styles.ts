const styles = {
  container: {
    width: '1060px',
    maxWidth: '95%',
    mt: '150px',
    mx: 'auto',
  },
  tabs: {
    display: 'grid',
    gridTemplateColumns: '60px 60px auto'
  },
  text: {
      color: 'white'
  }

};

export default styles;
