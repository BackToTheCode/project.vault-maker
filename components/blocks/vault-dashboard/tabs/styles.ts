const styles = {
  container: {
    maxWidth: '95%',
    // mt: '150px',
    mx: 'auto',
    width: '1060px'
  },
  tabs: {
    display: 'grid',
    gridTemplateColumns: '100px 100px auto',
    width: '100%'
  }
};

export default styles;
