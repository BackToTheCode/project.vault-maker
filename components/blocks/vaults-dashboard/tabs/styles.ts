const styles = {
  container: {
    maxWidth: '95%',
    mt: '150px',
    mx: 'auto',
    width: '1060px'
  },
  tabs: {
    display: 'grid',
    gridColumnGap: '20px',
    gridTemplateColumns: '70px 70px auto',

  }
};

export default styles;
