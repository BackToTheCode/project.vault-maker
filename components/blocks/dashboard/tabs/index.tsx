/** @jsx jsx */
import { jsx } from '@emotion/core';
import React, { Children, FC } from 'react';
import { Box, Flex, Text } from 'rebass';
import styles from './styles';

export interface TabsProps {
  children?: React.ReactNode | string;
}

export const Tabs: FC<TabsProps> = props => {
  const { children } = props;
  const { container, filler} = styles;
  return (
    <Flex sx={container}>
      <Box>
        <Text variant="body.regular">
          Open
        </Text>
        <Text variant="body.regular">
          Closed
        </Text>
      </Box>
      <Box sx={filler} />
    </Flex>
  );
};
