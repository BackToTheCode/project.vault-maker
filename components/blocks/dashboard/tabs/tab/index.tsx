import { FC } from 'react';
import { Box, Text } from 'rebass';
import styles from './styles';

export interface TabProps {
  children: React.ReactNode | string;
  active?: boolean;
}

export const Tab: FC<TabProps> = props => {
  const { active, children } = props;

  return (
    <Box sx={{...styles.container,...(active ? styles.container.active : {}) }}>
      <Text variant="body.regular" sx={{...styles.text, ...(active ? styles.active : {})}}>{children}</Text>
    </Box>
  );
};

Tab.displayName = 'Tab';
