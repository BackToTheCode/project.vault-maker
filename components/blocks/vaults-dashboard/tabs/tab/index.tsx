/** @jsx jsx */
import { jsx } from '@emotion/core';
import React, { Children, FC } from 'react';
import { Box, Flex, Text } from 'rebass';
import styles from './styles';

export interface TabProps {
  children?: React.ReactNode | string;
  isActive?: boolean;
}

// Create GridContainer from box

export const Tab: FC<TabProps> = props => {
  const { children, isActive } = props;
  const { active, text } = styles;
  return (
    <Text variant="body.regular" sx={{ ...text, ...(isActive ? active : {}) }}>
      {children}
    </Text>
  );
};
