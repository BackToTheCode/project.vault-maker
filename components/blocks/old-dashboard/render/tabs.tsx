import React from 'react';
import { Tabs } from '../tabs';

export const tabs = () => {
  return (
    <Tabs>
        <Tabs.Tab active={true}>Open</Tabs.Tab>
        <Tabs.Tab>Closed</Tabs.Tab>
    </Tabs>
  );
};
