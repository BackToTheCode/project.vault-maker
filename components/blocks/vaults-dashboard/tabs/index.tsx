/** @jsx jsx */
import { Container } from '@backtothecode/vm-ui-library';
import { jsx } from '@emotion/core';
import React, { FC } from 'react';
import { Box } from 'rebass';
import styles from './styles';
import { Tab } from './tab';

export interface TabsProps {
  sx?: any;
}

// Create GridContainer from box

export const Tabs: FC<TabsProps> = props => {
  const { sx } = props;
  const { container, tabs} = styles;
  return (
    <Container sx={{...container, ...sx}}>
      <Box sx={tabs}>
        <Tab isActive={true}>Open</Tab>
        <Tab>Closed</Tab>
        <Tab />
      </Box>
    </Container>
  );
};
