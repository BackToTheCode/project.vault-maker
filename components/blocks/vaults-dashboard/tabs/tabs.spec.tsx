import { render } from '@testing-library/react';
import React from 'react';
import { Tabs } from '../tabs';

describe('<Tabs /> spec', () => {
  test('it renders the component', () => {
    // Arrange
    const expectedText = 'Open';

    // Act
    const { getByText } = render(<Tabs />);

    // Assert
    expect(getByText(expectedText)).toMatchSnapshot();
  });
});
