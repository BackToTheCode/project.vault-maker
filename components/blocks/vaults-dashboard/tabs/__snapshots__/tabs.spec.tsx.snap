// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tabs /> spec it renders the component 1`] = `
<div
  class="css-158ar5q"
>
  Open
</div>
`;
